### ISSUE TITLE
Spotify token won't auto-refresh after 3600s (1hour) on 1.5.0 prerelease and nightly

### ISSUE THREAD

**Issue by ampedPF** [univrsal/tuna#62]
Hello!

So I updated from 1.4.2 to 1.5.0 in order to use my own api keys from Spotify following the tutorial on the obs forum.

I works fine for 1 hour, and it seems that token validity is set by default to 3600s by Spotify.
```console
= 2020.09.12 23:27 =
{
    "access_token": [EDITED],
    "expires_in": 3600,
    "scope": "user-modify-playback-state user-read-playback-state user-read-currently-playing",
    "token_type": "Bearer"
}
```
But now the plugin won't update the files 1 hour after refreshing manually the token on the Tuna settings window, in the Spotify tab.

I don't recall having this issue with version 1.4.2 but I can be wrong though.

Thank you again for your awesome work.

ampedPF

**Comment by univrsal** [comment]
Do you have the obs studio log for when that happened?

**Comment by ampedPF** [comment]
Sure I was waiting for the token to expire and extracting the related info from the log file (which is also attached)

[2020-09-12 21-53-40.txt](https://github.com/univrsal/tuna/files/5213343/2020-09-12.21-53-40.txt)
```
23:27:23.106: [tuna] Refreshing Spotify token
23:27:23.151: [tuna] Unknown error occured when querying Spotify-API: 401 (response: {
23:27:23.151:     "error": {
23:27:23.151:         "message": "The access token expired",
23:27:23.151:         "status": 401
23:27:23.151:     }
23:27:23.151: }
23:27:23.151: )
23:27:25.106: [tuna] Refreshing Spotify token
23:27:25.151: [tuna] Unknown error occured when querying Spotify-API: 401 (response: {
23:27:25.151:     "error": {
23:27:25.151:         "message": "The access token expired",
23:27:25.151:         "status": 401
23:27:25.151:     }
23:27:25.151: }
23:27:25.151: )
```
I manually refresh the token
```
23:27:26.103: [tuna] Spotify response: {
23:27:26.103:     "access_token": "REDACTED",
23:27:26.103:     "expires_in": 3600,
23:27:26.103:     "refresh_token": null,
23:27:26.103:     "scope": "user-modify-playback-state user-read-playback-state user-read-currently-playing",
23:27:26.103:     "token_type": "Bearer"
23:27:26.103: }
23:27:26.103: 
23:27:26.103: [tuna] Successfully logged in
```
One hour later
```
00:27:28.297: [tuna] Refreshing Spotify token
00:27:28.343: [tuna] Unknown error occured when querying Spotify-API: 401 (response: {
00:27:28.343:     "error": {
00:27:28.343:         "message": "The access token expired",
00:27:28.343:         "status": 401
00:27:28.343:     }
00:27:28.343: }
00:27:28.343: )
00:27:30.297: [tuna] Refreshing Spotify token
00:27:30.341: [tuna] Unknown error occured when querying Spotify-API: 401 (response: {
00:27:30.341:     "error": {
00:27:30.341:         "message": "The access token expired",
00:27:30.341:         "status": 401
00:27:30.341:     }
00:27:30.341: }
00:27:30.341: )
```

**Comment by univrsal** [comment]
I think the issues was just that it never actually refreshed it, I apparently removed that part and never put it back. Should be fixed in 08472b83783257f3189870aba29734938aa38110. The nightly build broke so there's no binaries except for linux currently. I don't know when I'll fix it but I might just wait for the new obs release, because it'll change some things with the VLC source.
